heroic-core/src/main/java/com/spotify/heroic/metric/LocalMetricManager.java

```diff
@@ -811,7 +846,7 @@ boolean isRetainQuotaViolated() {
             return retained.get() >= retainLimit;
         }
 
-        private static int getLeft(long limit, long current) {
+        private int getLeft(long limit, long current) {
```

Comment:
Why remove static? It uses no object state, right...?

Reply:
This was the reason the whole class was marked as static. There is no obvious reason why it should be static as the methods using this method called on instance rather than static context.

Reply:
This is a utility  method keep it static

Reply:
It's not possible to keep it static anymore. I've tested this change locally by sending tons of request and I've deployed this as individual container to production. None of the regular metrics - memory or stats were affected. 
We will have to observe for any anomalies but this is the best I can do in regards of verifying this change.